src/components/Nodes/GethConfig/ConfigForm.jsx

```diff
@@ -34,65 +25,71 @@ class ConfigForm extends Component {
   }
 
   componentDidMount() {
-    this.getConfig()
+    this.setDefaultConfig()
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    const { config } = this.state
-    const { config: prevConfig } = prevState
-    if (prevConfig !== config) {
-      const { dispatch } = this.props
-      dispatch(setConfig({ config }))
+  setDefaultConfig() {
+    // Set default config if no config set
+    const { client, dispatch } = this.props
+    const { config } = client
+    if (config.name) {
+      // Config already set
+      return
     }
+    const defaultConfig = geth.getConfig()
+    dispatch(setConfig({ config: defaultConfig }))
   }
 
-  async getConfig() {
-    const config = await geth.getConfig()
-    this.setState({ config })
-  }
-
-  handleChangeDataDir = dataDir => {
-    const { config } = this.state
+  handleChangeDataDir = event => {
+    const { dispatch, client } = this.props
+    const { config } = client
+    const dataDir = event.target.value
     const newConfig = { ...config, dataDir }
-    this.setState({ config: newConfig })
+    dispatch(setConfig({ config: newConfig }))
   }
 
   handleChangeSyncMode = syncMode => {
-    const { config } = this.state
+    const { client, dispatch } = this.props
+    const { config } = client
     const newConfig = { ...config, syncMode }
-    this.setState({ config: newConfig })
+    dispatch(setConfig({ config: newConfig }))
   }
 
   handleChangeIpc = ipc => {
-    const { config } = this.state
+    const { client, dispatch } = this.props
+    const { config } = client
     const newConfig = { ...config, ipc }
-    this.setState({ config: newConfig })
+    dispatch(setConfig({ config: newConfig }))
```

Comment:
nit: this batch of functions looks like good candidates to be DRYed up into a generalized function. can be implemented in a future PR.